src/cpp/cpp_typecheck_code.cpp

```diff
@@ -88,16 +88,15 @@ void cpp_typecheckt::typecheck_try_catch(codet &code)
         }
 
         // typecheck the body
-        typecheck_code(code);
+        typecheck_code(catch_block);
 
         // the declaration is now in a decl_block
-
-        assert(!code.operands().empty());
-        assert(to_code(code.op0()).get_statement()==ID_decl_block);
+        CHECK_RETURN(!catch_block.statements().empty());
+        CHECK_RETURN(catch_block.statements().front().get_statement() == ID_decl_block);
 
         // get the declaration
-        const code_declt &code_decl=
-          to_code_decl(to_code(code.op0().op0()));
+        const code_declt &code_decl =
+          to_code_decl(to_code(catch_block.statements().front().op0()));
```

Comment:
Use `to_code_block(...).statements().front()` instead of `op0()`?

Reply:
I don't think I understand? It's a "decl_block" (not a "block"), which really only the front-ends know about.

Reply:
OK, I took the `op0()` subject to `to_code_decl` to imply this was a `code_blockt`